src/main/java/teammates/common/datatransfer/StudentAttributes.java

```diff
@@ -130,7 +129,7 @@ public boolean isRegistered() {
     }
 
     public String getRegistrationUrl() {
-        return new Url(Config.APP_URL + Const.ActionURIs.STUDENT_COURSE_JOIN_NEW)
+        return new Url(Const.ActionURIs.STUDENT_COURSE_JOIN_NEW)
```

Comment:
I understand that the changes are necessary. But we can do something like what was done in #4546. Add the test cases in the new issue and we can merge this one soon. If you feel all the above can be done quickly, I guess there is no harm in it